pixel.ts

```diff
@@ -0,0 +1,75 @@
+namespace crickit {
+    const _CRICKIT_NEOPIXEL_PIN = 27
+
+    let brightness = 20;
+
+    /**
+     * Set the on-board pixel to a given color.
+     * @param color RGB color of the LED
+     */
+    //% group="NeoPixel"
+    //% blockId=sawsetpixelcolor block="crickit set pixel color %rgb=colorNumberPicker"
+    //% weight=99
+    //% blockGap=8
+    export function setColor(color: number): void {
+        if (brightness < 255)
+            color = fade(color, brightness);
+
+        let red = unpackR(color);
+        let green = unpackG(color);
+        let blue = unpackB(color);
+
+        const dev = saw();
+
+        const buffer: Buffer = pins.createBufferFromArray([
+            0,
+            5,
```

Comment:
this would write the data at an offset of 5 bytes into the internal neopixel buffer.
Would it make sense to change ```public neopixelSendBuffer(pin: number, buffer: Buffer)``` in the seesaw module to: ```public neopixelSendBuffer(pin: number, buffer: Buffer, offset:number=0)``` ?

then ```buffer``` would just be the color data and the seesaw library would add the offset if necessary.

Reply:
for larger buffers, we should avoid cloning it as we might run out of memory. so it's the responsability of the caller to set the offset right (not great but memory wins)

Reply:
now thinking about it, we're missing the 2 bytes for the i2c command so that's not good either.